kusto/query_builder.go

```diff
@@ -506,6 +507,173 @@ func NewStmt(query stringConstant, options ...StmtOption) Stmt {
+	if value == "" {
+		return value
+	}
+
+	var literal strings.Builder
+
+	if hidden {
+		literal.WriteString("h")
+	}
+	literal.WriteString("\"")
+	for _, c := range value {
+		switch c {
+		case '\'':
+			literal.WriteString("\\'")
+
+		case '"':
+			literal.WriteString("\\\"")
+
+		case '\\':
+			literal.WriteString("\\\\")
+
+		case '\x00':
+			literal.WriteString("\\0")
+
+		case '\a':
+			literal.WriteString("\\a")
+
+		case '\b':
+			literal.WriteString("\\b")
+
+		case '\f':
+			literal.WriteString("\\f")
+
+		case '\n':
+			literal.WriteString("\\n")
+
+		case '\r':
+			literal.WriteString("\\r")
+
+		case '\t':
+			literal.WriteString("\\t")
+
+		case '\v':
+			literal.WriteString("\\v")
+
+		default:
+			if !ShouldBeEscaped(c) {
+				literal.WriteString(string(c))
+			} else {
+				literal.WriteString(fmt.Sprintf("\\u%x", c))
+			}
+
+		}
+	}
+	literal.WriteString("\"")
+
+	return literal.String()
+}
+
+// ShouldBeEscaped Checks whether a rune should be escaped or not based on it's type.
+func ShouldBeEscaped(c int32) bool {
+	if c <= unicode.MaxLatin1 {
+		return unicode.IsControl(c)
+	}
+	return true
+}
+
+// AddInt will add an int as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddInt(query int) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%d", query)
+	return s
+}
+
+// AddFloat32 will add a Float32 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddFloat32(query float32) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%f", query)
+	return s
+}
+
+// AddFloat64 will add a Float64 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddFloat64(query float64) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%f", query)
+	return s
+}
+
+// AddComplex64 will add a Complex64 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddComplex64(query complex64) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%v", query)
+	return s
+}
+
+// AddComplex128 will add a Complex128 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddComplex128(query complex128) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%v", query)
+	return s
+}
+
+// AddBool will add a bool as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddBool(query bool) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%t", query)
+	return s
+}
+
+// AddByte will add a byte as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddByte(query byte) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%d", query)
+	return s
+}
+
+// AddRune will add a rune as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddRune(query rune) Stmt {
+	s.queryStr = s.queryStr + fmt.Sprintf("%c", query)
```

Comment:
If this is a unicode char it should be quoted, otherwise KQL will mistreat it.

Reply:
Rune is all chars, but yeah we should still escape it with the same logic as string.

Reply:
And you probably need to go through AddQuotedString to make sure the char is properly escaped.

Reply:
On third thought, I don't see a usable scenario for this API?

Reply:
So should we keep them? both rune and byte?

Reply:
per our discussion, let's remove this API completely